diff -ru a/drivers/net/virtio/virtio_rxtx.c b/drivers/net/virtio/virtio_rxtx.c
--- a/drivers/net/virtio/virtio_rxtx.c	2015-08-17 19:35:37.000000000 +0200
+++ b/drivers/net/virtio/virtio_rxtx.c	2016-02-11 16:16:51.448162515 +0100
@@ -506,9 +506,6 @@
 	if (likely(num > DESC_PER_CACHELINE))
 		num = num - ((rxvq->vq_used_cons_idx + num) % DESC_PER_CACHELINE);
 
-	if (num == 0)
-		return 0;
-
 	num = virtqueue_dequeue_burst_rx(rxvq, rcv_pkts, len, num);
 	PMD_RX_LOG(DEBUG, "used:%d dequeue:%d", nb_used, num);
 
@@ -601,9 +598,6 @@
 
 	virtio_rmb();
 
-	if (nb_used == 0)
-		return 0;
-
 	PMD_RX_LOG(DEBUG, "used:%d\n", nb_used);
 
 	hw = rxvq->hw;
