diff -ru a/rx_pkt.c b/rx_pkt.c
--- a/rx_pkt.c	2015-08-18 19:31:30.000000000 +0200
+++ b/rx_pkt.c	2016-02-29 04:24:12.563895767 +0100
@@ -89,7 +89,7 @@
 	START_EMPTY_MEASSURE();
 	*mbufs = tbase->ws_mbuf->mbuf[0] +
 		(RTE_ALIGN_CEIL(tbase->ws_mbuf->idx[0].prod, 2) & WS_MBUF_MASK);
-	lr = tbase->rx_params_sw.last_read_ring;
+	lr = tbase->rx_params_hw.last_read_portid;
 	nb_rx = rte_eth_rx_burst(tbase->rx_params_hw.rx_pq[lr].port,
 				 tbase->rx_params_hw.rx_pq[lr].queue,
 				 *mbufs, MAX_PKT_BURST);
@@ -114,7 +114,6 @@
 
 static inline uint16_t rx_pkt_hw1_param(struct task_base *tbase, struct rte_mbuf ***mbufs, int twice)
 {
-	uint8_t lr;
 	uint16_t nb_rx;
 
 	START_EMPTY_MEASSURE();
@@ -194,7 +193,7 @@
 		lr = lr + 1 == tbase->rx_params_sw.nb_rxrings? 0 : lr + 1;
 	} while(!nb_rx && lr != tbase->rx_params_sw.last_read_ring);
 
-
+	tbase->rx_params_sw.last_read_ring = lr;
 
 	if (nb_rx != 0) {
 		TASK_STATS_ADD_RX(&tbase->aux->stats, nb_rx);
