diff -ru a/defaults.c b/defaults.c
--- a/defaults.c	2015-09-14 14:43:30.000000000 +0200
+++ b/defaults.c	2016-01-07 15:45:20.365893835 +0100
@@ -47,8 +47,8 @@
 #define NB_PIPES        32768
 #define NB_MBUF         4096
 #define RING_RX_SIZE    256
-#define NB_RX_RING_DESC 256
-#define NB_TX_RING_DESC 256
+#define NB_RX_RING_DESC 2048
+#define NB_TX_RING_DESC 2048
 
 /* 1500000 milliseconds */
 #define DEFAULT_CPE_TIMEOUT_MS    1500000
@@ -170,7 +170,7 @@
 
 			targ->n_max_rules = 1024;
 			targ->ring_size = RING_RX_SIZE;
-			targ->nb_cache_mbuf = MAX_PKT_BURST * 4;
+			targ->nb_cache_mbuf = MAX_PKT_BURST * 8;
 			targ->overhead = ETHER_CRC_LEN + 20;
 
 			targ->tunnel_hop_limit = 3;
