diff -ru a/prox-configs/handle_5tuplookup-4.cfg b/prox-configs/handle_5tuplookup-4.cfg
--- a/prox-configs/handle_5tuplookup-4.cfg	2015-09-23 04:31:45.000000000 +0200
+++ b/prox-configs/handle_5tuplookup-4.cfg	2016-01-06 13:50:20.253185197 +0100
@@ -33,7 +33,7 @@
 mac=50:00:00:00:00:04
 
 [defaults]
-mempool size=4K
+mempool size=13K
 
 [global]
 start time=5
diff -ru a/prox-configs/handle_acl-4.cfg b/prox-configs/handle_acl-4.cfg
--- a/prox-configs/handle_acl-4.cfg	2015-09-23 04:31:45.000000000 +0200
+++ b/prox-configs/handle_acl-4.cfg	2016-01-06 13:51:47.284324105 +0100
@@ -34,7 +34,7 @@
 [lua]
 acl_table = dofile("/tmp/acl_rules-2.lua")
 [defaults]
-mempool size=65K
+mempool size=69K
 qinq tag=0x0081
 [global]
 start time=5
diff -ru a/prox-configs/handle_bng-4.cfg b/prox-configs/handle_bng-4.cfg
--- a/prox-configs/handle_bng-4.cfg	2015-09-23 04:31:45.000000000 +0200
+++ b/prox-configs/handle_bng-4.cfg	2016-01-06 13:55:35.466066451 +0100
@@ -44,7 +44,7 @@
 ;$wk=3s0-8s0,3s0h-8s0h; 12 workers
 
 [defaults]
-mempool size=16K
+mempool size=32K
 qinq tag=0x0081
 
 [global]
diff -ru a/prox-configs/handle_bng_qos-4.cfg b/prox-configs/handle_bng_qos-4.cfg
--- a/prox-configs/handle_bng_qos-4.cfg	2015-09-23 04:31:45.000000000 +0200
+++ b/prox-configs/handle_bng_qos-4.cfg	2016-01-06 13:56:28.904537736 +0100
@@ -39,7 +39,7 @@
 ;$wk=7s0-10s0,7s0h-10s0h; 8 workers
 
 [defaults]
-mempool size=128K
+mempool size=142K
 qinq tag=0x0081
 [lua]
 lpm4 = dofile("/tmp/ipv4.lua")
diff -ru a/prox-configs/handle_latency-1.cfg b/prox-configs/handle_latency-1.cfg
--- a/prox-configs/handle_latency-1.cfg	2015-09-25 19:16:22.000000000 +0200
+++ b/prox-configs/handle_latency-1.cfg	2016-01-06 13:53:16.379442592 +0100
@@ -27,7 +27,7 @@
 mac=50:00:00:00:00:02
 
 [defaults]
-mempool size=1875K
+mempool size=1879K
 
 [global]
 start time=5
diff -ru a/prox-configs/handle_none-4.cfg b/prox-configs/handle_none-4.cfg
--- a/prox-configs/handle_none-4.cfg	2015-09-23 04:31:45.000000000 +0200
+++ b/prox-configs/handle_none-4.cfg	2016-01-06 13:47:42.447746531 +0100
@@ -33,7 +33,7 @@
 mac=50:00:00:00:00:04
 
 [defaults]
-mempool size=4K
+mempool size=8K
 
 [global]
 start time=5
diff -ru a/prox-configs/handle_tag_untag-4.cfg b/prox-configs/handle_tag_untag-4.cfg
--- a/prox-configs/handle_tag_untag-4.cfg	2015-09-23 04:31:45.000000000 +0200
+++ b/prox-configs/handle_tag_untag-4.cfg	2016-01-06 13:49:07.517904850 +0100
@@ -33,7 +33,7 @@
 mac=50:00:00:00:00:04
 
 [defaults]
-mempool size=4K
+mempool size=8K
 
 [global]
 start time=5
diff -ru a/prox-configs/handle_touch-4.cfg b/prox-configs/handle_touch-4.cfg
--- a/prox-configs/handle_touch-4.cfg	2015-09-23 04:31:45.000000000 +0200
+++ b/prox-configs/handle_touch-4.cfg	2016-01-06 13:48:25.247323077 +0100
@@ -33,7 +33,7 @@
 mac=50:00:00:00:00:04
 
 [defaults]
-mempool size=4K
+mempool size=8K
 
 [global]
 start time=5
