diff -ru a/prox-configs/handle_bng_qos-4.cfg b/prox-configs/handle_bng_qos-4.cfg
--- a/prox-configs/handle_bng_qos-4.cfg	2015-09-23 04:31:45.000000000 +0200
+++ b/prox-configs/handle_bng_qos-4.cfg	2016-03-07 14:10:39.464246547 +0100
@@ -71,6 +71,7 @@
 untag mpls=yes
 network side=yes
 tx cores=$wk task=1 proto=ipv4
+drop=no
 
 [core 2s0]
 name=LB-cpe
@@ -89,6 +90,7 @@
 untag mpls=yes
 rx port=inet1
 tx cores=$wk task=1 proto=ipv4
+drop=no
 
 ;*****************************************************************************************
 ;#### Workers receiving from LB
@@ -111,6 +113,7 @@
 mode=qinqencapv4
 rx ring=yes
 tx cores from cpe table=3s0,4s0 remap=cpe0,cpe1 ;map packets going to cpe0 to 3s0 and cpe1 to 4s0
+drop=no
 classify=yes
 dscp=dscp_table
 user table=user_table
