diff -ru a/test_OVS_port_fwd.py b/test_OVS_port_fwd.py
--- a/test_OVS_port_fwd.py	2015-09-28 19:57:11.000000000 +0200
+++ b/test_OVS_port_fwd.py	2016-01-28 10:49:56.779716923 +0100
@@ -24,11 +24,7 @@
 from dats.remote_control import remote_system
 
 class ForwardPacketsNoTouch(dats.test.binsearch.BinarySearch):
-    """Port forwarding without touching packets
-
-    The application will take packets in from one port and forward them
-    unmodified to another port.  This use case is not representing any *real*
-    use case but it is a good start to do a sanity check of the environment.
+    """Port forwarding by OVS (PHY-OVS-PHY)
 
     The KPI is the number of packets per second for 64 byte packets with an
     accepted minimal packet loss.
@@ -47,7 +43,7 @@
 
     def setup_class(self):
         self._tester = self.get_remote('tester').run_prox_with_config("gen_all-4.cfg", "-e -t", "Tester")
-        self._sut = self.get_remote('sut').run_prox_with_config("handle_none-4.cfg", "-t", "SUT")
+        # OVS flows on SUT should have been properly setup, there is no need to run PROX on SUT
 
     def teardown_class(self):
         pass
