diff -ru a/prox-configs/gen_all-4.cfg b/prox-configs/gen_all-4.cfg
--- a/prox-configs/gen_all-4.cfg	2015-09-23 04:31:45.000000000 +0200
+++ b/prox-configs/gen_all-4.cfg	2016-01-29 09:19:47.893032963 +0100
@@ -42,7 +42,7 @@
 [core 0s0]
 mode=master
 
-[core 1s0]
+[core 1s1]
 name=cpe0
 task=0
 mode=gen
@@ -50,7 +50,7 @@
 bps=1250000000
 pkt inline=50 00 00 00 00 01 70 00 00 00 00 01 08 00 45 00 00 1c 00 01 00 00 40 11 f7 7d c0 a8 01 01 c0 a8 01 01 13 88 13 88 00 08 55 7b
 
-[core 2s0]
+[core 2s1]
 name=cpe0
 task=0
 mode=gen
@@ -58,7 +58,7 @@
 bps=1250000000
 pkt inline=50 00 00 00 00 02 70 00 00 00 00 02 08 00 45 00 00 1c 00 01 00 00 40 11 f7 7d c0 a8 01 01 c0 a8 01 01 13 88 13 88 00 08 55 7b
 
-[core 3s0]
+[core 3s1]
 name=cpe0
 task=0
 mode=gen
@@ -66,7 +66,7 @@
 bps=1250000000
 pkt inline=50 00 00 00 00 03 70 00 00 00 00 03 08 00 45 00 00 1c 00 01 00 00 40 11 f7 7d c0 a8 01 01 c0 a8 01 01 13 88 13 88 00 08 55 7b
 
-[core 4s0]
+[core 4s1]
 name=cpe0
 task=0
 mode=gen
diff -ru a/test_OVS_port_fwd.py b/test_OVS_port_fwd.py
--- a/test_OVS_port_fwd.py	2016-01-28 11:03:26.559102367 +0100
+++ b/test_OVS_port_fwd.py	2016-01-29 09:29:48.491388043 +0100
@@ -49,7 +49,7 @@
         pass
 
     def run_test(self, pkt_size, duration, value):
-        cores = [1, 2, 3, 4]
+        cores = [15, 16, 17, 18]
 
         self._tester.stop_all()
         self._tester.reset_stats()
