diff -ru a/prox-configs/gen_5tuplookup-4.cfg b/prox-configs/gen_5tuplookup-4.cfg
--- a/prox-configs/gen_5tuplookup-4.cfg	2015-09-23 04:31:45.000000000 +0200
+++ b/prox-configs/gen_5tuplookup-4.cfg	2016-02-12 05:00:01.271218460 +0100
@@ -43,7 +43,7 @@
 [core 0s0]
 mode=master
 
-[core 1s0]
+[core 1s1]
 name=p0
 task=0
 mode=gen
@@ -59,7 +59,7 @@
 random=00000000000XXXXX00000000000XXXXX
 rand_offset=34
 
-[core 2s0]
+[core 2s1]
 name=p1
 task=0
 mode=gen
@@ -75,7 +75,7 @@
 random=00000000000XXXXX00000000000XXXXX
 rand_offset=34
 
-[core 3s0]
+[core 3s1]
 name=p2
 task=0
 mode=gen
@@ -91,7 +91,7 @@
 random=00000000000XXXXX00000000000XXXXX
 rand_offset=34
 
-[core 4s0]
+[core 4s1]
 name=p3
 task=0
 mode=gen
diff -ru a/prox-configs/gen_acl-4.cfg b/prox-configs/gen_acl-4.cfg
--- a/prox-configs/gen_acl-4.cfg	2015-09-23 04:31:45.000000000 +0200
+++ b/prox-configs/gen_acl-4.cfg	2016-02-12 05:01:08.200562172 +0100
@@ -45,7 +45,7 @@
 [core 0s0]
 mode=master
 
-[core 1s0]
+[core 1s1]
 name=p0
 task=0
 mode=gen
@@ -77,7 +77,7 @@
 lat=yes
 lat pos=50
 
-[core 2s0]
+[core 2s1]
 name=p1
 task=0
 mode=gen
@@ -109,7 +109,7 @@
 lat=yes
 lat pos=50
 
-[core 3s0]
+[core 3s1]
 name=p2
 task=0
 mode=gen
@@ -141,7 +141,7 @@
 lat=yes
 lat pos=50
 
-[core 4s0]
+[core 4s1]
 name=p3
 task=0
 mode=gen
diff -ru a/prox-configs/gen_all-4.cfg b/prox-configs/gen_all-4.cfg
--- a/prox-configs/gen_all-4.cfg	2015-09-23 04:31:45.000000000 +0200
+++ b/prox-configs/gen_all-4.cfg	2016-02-12 05:01:58.004073986 +0100
@@ -42,7 +42,7 @@
 [core 0s0]
 mode=master
 
-[core 1s0]
+[core 1s1]
 name=cpe0
 task=0
 mode=gen
@@ -50,7 +50,7 @@
 bps=1250000000
 pkt inline=50 00 00 00 00 01 70 00 00 00 00 01 08 00 45 00 00 1c 00 01 00 00 40 11 f7 7d c0 a8 01 01 c0 a8 01 01 13 88 13 88 00 08 55 7b
 
-[core 2s0]
+[core 2s1]
 name=cpe0
 task=0
 mode=gen
@@ -58,7 +58,7 @@
 bps=1250000000
 pkt inline=50 00 00 00 00 02 70 00 00 00 00 02 08 00 45 00 00 1c 00 01 00 00 40 11 f7 7d c0 a8 01 01 c0 a8 01 01 13 88 13 88 00 08 55 7b
 
-[core 3s0]
+[core 3s1]
 name=cpe0
 task=0
 mode=gen
@@ -66,7 +66,7 @@
 bps=1250000000
 pkt inline=50 00 00 00 00 03 70 00 00 00 00 03 08 00 45 00 00 1c 00 01 00 00 40 11 f7 7d c0 a8 01 01 c0 a8 01 01 13 88 13 88 00 08 55 7b
 
-[core 4s0]
+[core 4s1]
 name=cpe0
 task=0
 mode=gen
diff -ru a/prox-configs/gen_bng-4.cfg b/prox-configs/gen_bng-4.cfg
--- a/prox-configs/gen_bng-4.cfg	2016-01-06 12:57:07.068781696 +0100
+++ b/prox-configs/gen_bng-4.cfg	2016-02-12 05:04:45.490435638 +0100
@@ -42,7 +42,7 @@
 [core 0s0]
 mode=master
 
-[core 1s0]
+[core 1s1]
 name=arp
 
 task=0
@@ -66,7 +66,7 @@
 rand_offset=18
 
 
-[core 2s0]
+[core 2s1]
 name=cpe
 task=0
 mode=gen
@@ -85,7 +85,7 @@
 lat=yes
 lat pos=42
 
-[core 4s0]
+[core 4s1]
 name=cpe
 task=0
 mode=gen
@@ -105,7 +105,7 @@
 lat=yes
 lat pos=42
 
-[core 5s0]
+[core 5s1]
 name=inet0
 task=0
 mode=gen
@@ -119,7 +119,7 @@
 lat=yes
 lat pos=66
 
-[core 6s0]
+[core 6s1]
 name=inet1
 task=0
 mode=gen
@@ -132,28 +132,28 @@
 lat=yes
 lat pos=66
 
-[core 7s0]
+[core 7s1]
 name=CPE0
 task=0
 mode=lat
 rx port=cpe0
 lat pos=42
 
-[core 8s0]
+[core 8s1]
 name=CPE1
 task=0
 mode=lat
 rx port=cpe1
 lat pos=42
 
-[core 9s0]
+[core 9s1]
 name=INET0
 task=0
 mode=lat
 rx port=inet0
 lat pos=66
 
-[core 3s0]
+[core 3s1]
 name=INET1
 task=0
 mode=lat
diff -ru a/prox-configs/gen_bng_qos-4.cfg b/prox-configs/gen_bng_qos-4.cfg
--- a/prox-configs/gen_bng_qos-4.cfg	2016-01-06 12:57:07.069781686 +0100
+++ b/prox-configs/gen_bng_qos-4.cfg	2016-02-12 05:05:58.242723956 +0100
@@ -42,7 +42,7 @@
 [core 0s0]
 mode=master
 
-[core 1s0]
+[core 1s1]
 name=arp
 
 task=0
@@ -66,7 +66,7 @@
 rand_offset=18
 
 
-[core 2s0]
+[core 2s1]
 name=cpe
 task=0
 mode=gen
@@ -85,7 +85,7 @@
 lat=yes
 lat pos=42
 
-[core 4s0]
+[core 4s1]
 name=cpe
 task=0
 mode=gen
@@ -105,7 +105,7 @@
 lat=yes
 lat pos=42
 
-[core 5s0]
+[core 5s1]
 name=inet0
 task=0
 mode=gen
@@ -119,7 +119,7 @@
 lat=yes
 lat pos=66
 
-[core 6s0]
+[core 6s1]
 name=inet1
 task=0
 mode=gen
@@ -132,28 +132,28 @@
 lat=yes
 lat pos=66
 
-[core 7s0]
+[core 7s1]
 name=CPE0
 task=0
 mode=lat
 rx port=cpe0
 lat pos=42
 
-[core 8s0]
+[core 8s1]
 name=CPE1
 task=0
 mode=lat
 rx port=cpe1
 lat pos=42
 
-[core 9s0]
+[core 9s1]
 name=INET0
 task=0
 mode=lat
 rx port=inet0
 lat pos=66
 
-[core 3s0]
+[core 3s1]
 name=INET1
 task=0
 mode=lat
diff -ru a/prox-configs/gen_latency-1.cfg b/prox-configs/gen_latency-1.cfg
--- a/prox-configs/gen_latency-1.cfg	2015-09-23 04:31:45.000000000 +0200
+++ b/prox-configs/gen_latency-1.cfg	2016-02-12 05:06:18.956521323 +0100
@@ -42,7 +42,7 @@
 [core 0s0]
 mode=master
 
-[core 1s0]
+[core 1s1]
 name=cpe0
 task=0
 mode=gen
@@ -52,7 +52,7 @@
 lat pos=42
 lat=yes
 
-[core 2s0]
+[core 2s1]
 task=0
 mode=lat
 rx port=p1
diff -ru a/prox-configs/gen_tag_untag-4.cfg b/prox-configs/gen_tag_untag-4.cfg
--- a/prox-configs/gen_tag_untag-4.cfg	2015-09-23 04:31:45.000000000 +0200
+++ b/prox-configs/gen_tag_untag-4.cfg	2016-02-12 05:09:59.818360677 +0100
@@ -43,7 +43,7 @@
 [core 0s0]
 mode=master
 
-[core 1s0]
+[core 1s1]
 name=tag1
 task=0
 mode=gen
@@ -53,7 +53,7 @@
 lat=yes
 lat pos=46
 
-[core 2s0]
+[core 2s1]
 name=udp1
 task=0
 mode=gen
@@ -63,7 +63,7 @@
 lat=yes
 lat pos=42
 
-[core 3s0]
+[core 3s1]
 name=udp2
 task=0
 mode=gen
@@ -73,7 +73,7 @@
 lat=yes
 lat pos=42
 
-[core 4s0]
+[core 4s1]
 name=tag2
 task=0
 mode=gen
diff -ru a/test_01_port_fwd.py b/test_01_port_fwd.py
--- a/test_01_port_fwd.py	2015-09-28 19:57:11.000000000 +0200
+++ b/test_01_port_fwd.py	2016-02-12 05:11:12.776646912 +0100
@@ -53,7 +53,7 @@
         pass
 
     def run_test(self, pkt_size, duration, value):
-        cores = [1, 2, 3, 4]
+        cores = [15, 16, 17, 18]
 
         self._tester.stop_all()
         self._tester.reset_stats()
diff -ru a/test_02_pkt_touch.py b/test_02_pkt_touch.py
--- a/test_02_pkt_touch.py	2015-09-28 22:04:11.000000000 +0200
+++ b/test_02_pkt_touch.py	2016-02-12 05:11:51.572267359 +0100
@@ -54,7 +54,7 @@
         pass
 
     def run_test(self, pkt_size, duration, value):
-        cores = [1, 2, 3, 4]
+        cores = [15, 16, 17, 18]
 
         self._tester.stop_all()
         self._tester.reset_stats()
diff -ru a/test_03_tag_untag.py b/test_03_tag_untag.py
--- a/test_03_tag_untag.py	2015-09-28 22:14:10.000000000 +0200
+++ b/test_03_tag_untag.py	2016-02-12 05:12:52.555670727 +0100
@@ -57,9 +57,9 @@
         pass
 
     def run_test(self, pkt_size, duration, value):
-        cores_tagged = [1, 4]
-        cores_plain = [2, 3]
-        cores = [1, 2, 3, 4]
+        cores_tagged = [15, 18]
+        cores_plain = [16, 17]
+        cores = [15, 16, 17, 18]
 
         self._tester.stop_all()
         self._tester.reset_stats()
diff -ru a/test_04_5-tuple_lookup.py b/test_04_5-tuple_lookup.py
--- a/test_04_5-tuple_lookup.py	2015-09-28 22:23:03.000000000 +0200
+++ b/test_04_5-tuple_lookup.py	2016-02-12 05:13:29.122312972 +0100
@@ -58,7 +58,7 @@
         pass
 
     def run_test(self, pkt_size, duration, value):
-        cores = [1, 2, 3, 4]
+        cores = [15, 16, 17, 18]
 
         self._tester.stop_all()
         self._tester.reset_stats()
diff -ru a/test_05_acl.py b/test_05_acl.py
--- a/test_05_acl.py	2015-09-28 22:35:40.000000000 +0200
+++ b/test_05_acl.py	2016-02-12 05:14:09.562917312 +0100
@@ -59,7 +59,7 @@
         pass
 
     def run_test(self, pkt_size, duration, value):
-        cores = [1, 2, 3, 4]
+        cores = [15, 16, 17, 18]
 
         self._tester.stop_all()
         self._tester.reset_stats()
diff -ru a/test_06_buffering-and-latency.py b/test_06_buffering-and-latency.py
--- a/test_06_buffering-and-latency.py	2015-09-28 23:12:52.000000000 +0200
+++ b/test_06_buffering-and-latency.py	2016-02-12 05:14:56.339459659 +0100
@@ -59,8 +59,8 @@
         pass
 
     def run_test(self, pkt_size, duration, value):
-        core_tx = 1
-        core_rx = 2
+        core_tx = 15
+        core_rx = 16
 
         self._tester.stop_all()
         self._tester.reset_stats()
diff -ru a/test_07_bng.py b/test_07_bng.py
--- a/test_07_bng.py	2015-09-29 00:11:20.000000000 +0200
+++ b/test_07_bng.py	2016-02-12 05:17:45.520804368 +0100
@@ -60,11 +60,11 @@
 
         # These should go to the configuration file and eventually should be
         # autoprobed.
-        self._cpe_cores = [2, 4]
-        self._inet_cores = [5, 6]
-        self._arp_cores = [1]
-        self._all_rx_cores = [3, 7, 8, 9]
-        self._rx_lat_cores = [3, 7, 8, 9]
+        self._cpe_cores = [16, 18]
+        self._inet_cores = [19, 20]
+        self._arp_cores = [15]
+        self._all_rx_cores = [17, 21, 22, 23]
+        self._rx_lat_cores = [17, 21, 22, 23]
         self._all_stats_cores = self._cpe_cores + self._inet_cores + self._all_rx_cores
 
         self._step_delta = 1
diff -ru a/test_08_bng_qos.py b/test_08_bng_qos.py
--- a/test_08_bng_qos.py	2015-09-29 00:33:36.000000000 +0200
+++ b/test_08_bng_qos.py	2016-02-12 05:19:24.872831827 +0100
@@ -61,11 +61,11 @@
 
         # These should go to the configuration file and eventually should be
         # autoprobed.
-        self._cpe_cores = [2, 4]
-        self._inet_cores = [5, 6]
-        self._arp_cores = [1]
-        self._all_rx_cores = [3, 7, 8, 9]
-        self._rx_lat_cores = [3, 7, 8, 9]
+        self._cpe_cores = [16, 18]
+        self._inet_cores = [19, 20]
+        self._arp_cores = [15]
+        self._all_rx_cores = [17, 21, 22, 23]
+        self._rx_lat_cores = [17, 21, 22, 23]
         self._all_stats_cores = self._cpe_cores + self._inet_cores + self._all_rx_cores
 
         self._step_delta = 1
